modules/View.js

```diff
@@ -748,7 +782,11 @@
           }
         }
       }
-      return prepareFieldsContext;
+      var context = this._prepare(prepareFieldsContext);
+      if (!_.isUndefined(context)) {
```

Comment:
if _prepare() returns undefined then prepareFieldsContext is ignored.  I don't think that is what you want?

Please add tests for this so that we know it is working correctly for the corner cases and that it continues to work correctly :).

Reply:
do the tests run for you? it breaks when i run them

Reply:
also, good catch.